src/Altinn.Notifications.Integrations/Register/RegisterClient.cs

```diff
@@ -1,52 +1,136 @@
+    /// <returns>
+    /// A task that represents the asynchronous operation.
+    /// The task result contains a list of <see cref="OrganizationContactPoints"/> representing the contact points of the specified organizations.
+    /// </returns>
+    public async Task<List<OrganizationContactPoints>> GetOrganizationContactPoints(List<string> organizationNumbers)
+    {
+        if (organizationNumbers == null || organizationNumbers.Count == 0)
         {
-            _client = client;
-            _client.BaseAddress = new Uri(settings.Value.ApiRegisterEndpoint);
+            return [];
         }
 
-        /// <inheritdoc/>
-        public async Task<List<OrganizationContactPoints>> GetOrganizationContactPoints(List<string> organizationNumbers)
+        var lookupObject = new OrgContactPointLookup
         {
-            var lookupObject = new OrgContactPointLookup
-            {
-                OrganizationNumbers = organizationNumbers
-            };
+            OrganizationNumbers = organizationNumbers
+        };
 
-            HttpContent content = new StringContent(JsonSerializer.Serialize(lookupObject, JsonSerializerOptionsProvider.Options), Encoding.UTF8, "application/json");
+        HttpContent content = new StringContent(JsonSerializer.Serialize(lookupObject, _jsonSerializerOptions), Encoding.UTF8, "application/json");
 
-            var response = await _client.PostAsync("organizations/contactpoint/lookup", content);
+        var response = await _client.PostAsync(_contactPointLookupEndpoint, content);
 
-            if (!response.IsSuccessStatusCode)
-            {
-                throw await PlatformHttpException.CreateAsync(response);
-            }
+        if (!response.IsSuccessStatusCode)
+        {
+            throw await PlatformHttpException.CreateAsync(response);
+        }
+
+        string responseContent = await response.Content.ReadAsStringAsync();
+        var contactPoints = JsonSerializer.Deserialize<OrgContactPointsList>(responseContent, _jsonSerializerOptions)?.ContactPointsList;
+        return contactPoints ?? [];
+    }
 
-            string responseContent = await response.Content.ReadAsStringAsync();
-            List<OrganizationContactPoints>? contactPoints = JsonSerializer.Deserialize<OrgContactPointsList>(responseContent, JsonSerializerOptionsProvider.Options)!.ContactPointsList;
-            return contactPoints!;
+    /// <summary>
+    /// Asynchronously retrieves party details for the specified organizations.
+    /// </summary>
+    /// <param name="organizationNumbers">A collection of organization numbers for which party details are requested.</param>
+    /// <returns>
+    /// A task that represents the asynchronous operation.
+    /// The task result contains a list of <see cref="PartyDetails"/> representing the details of the specified organizations.
+    /// </returns>
+    public async Task<List<PartyDetails>> GetPartyDetailsForOrganizations(List<string> organizationNumbers)
```

Comment:
Names lookup allows a mix of organization and national identity numbers in the same request. I think splitting them in separate methods is unnecessary.